### PR TITLE
Movie grid test

### PR DESCRIPTION
**Pull request by reshav-abraham**

### CODE REVIEW

**Comment by reshav-abraham** [comment]
Hey Frank! 
Wanted to get a better understanding of Container and Presentational Components. At my old job, not too much thought was done in designing the code this way. We just had reusable components but not to this degree. Also do you guys use function based components or class components?

I created a Movie Grid container component and Grid Presentational component. Lmk what you think!

**Comment by fkotsian** [comment]
> Hey Frank!
> Wanted to get a better understanding of Container and Presentational Components. At my old job, not too much thought was done in designing the code this way. We just had reusable components but not to this degree. Also do you guys use function based components or class components?
> 
> I created a Movie Grid container component and Grid Presentational component. Lmk what you think!

Nice! Great questions. Let me leave you some answers here and will leave some comments on the PR as well:

> Wanted to get a better understanding of Container and Presentational Components. At my old job, not too much thought was done in designing the code this way. 

Def! We are definitely in-flight ourselves, but moving towards a system of using:
- Container components to fetch & pass down data 
- Smaller Presentational components to accept data props and render them
  - Presentational components can also have sub-components - no need to go crazy with that, we certainly don't lol - but the key thing is that they get data as props

Basically data-fetcher Containers vs data-presenting Presenters 

> Also do you guys use function based components or class components?

We have both right now but are moving to Hooks over componentDidMount and so forth. I would say roll with function-based components, but if something lends itself much more easily to class components, that is 100% fine. We definitely have code that is moving over to functions/Hooks

File: client/components/Grid.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React, { useState, useEffect } from 'react';
+import {AgGridColumn, AgGridReact} from 'ag-grid-react';
+import 'ag-grid-community/dist/styles/ag-grid.css';
+import 'ag-grid-community/dist/styles/ag-theme-alpine.css';
```
Comment (fkotsian) [inline client/components/Grid.js:4]: Ai crap, I forgot to mention that we use [Semantic UI](https://semantic-ui.com/collections/grid.html) in a lot of places in our legacy apps (our new apps are moving towards our internal design system). Might help you get familiar with it to use it here as well? Completely up to you though

Here's the Grid spec for example: https://semantic-ui.com/collections/grid.html

From an exercise POV though, definitely like the AgGrid/AgColumn architecture here, esp field names. Very clean

File: client/components/Grid.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+/*
+Presentation Component For Grid data
+Pass row columns and data
+*/
+
+
+export const Grid = props => {
```
Comment (fkotsian) [inline client/components/Grid.js:12]: like it, functional component

File: client/components/MovieGrid.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            data.results.forEach((x, i) => movieData.push({"Movie": x.title, "Overview": x.overview, "Release Date": x.release_date, "Language": x.original_language}));
+            setMovies(movieData);
+        })
+        .catch(err => console.log(err));
+        
+    },[])
+
+   return (
+       <div>
+            <Grid rowData={movies} fields={["Movie", "Overview", "Release Date", "Language"]}/>
```
Comment (fkotsian) [inline client/components/MovieGrid.js:30]: `fields` 👍

File: client/components/MovieGrid.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+export const MovieGrid = () => {
+    const [movies, setMovies] = useState([]);
+
+    useEffect(()=> {
+        // console.log(getTopMovies());
+        let movieData = [];
+        getTopMovies().then(data => {
+            data.results.forEach((x, i) => movieData.push({"Movie": x.title, "Overview": x.overview, "Release Date": x.release_date, "Language": x.original_language}));
+            setMovies(movieData);
```
Comment (fkotsian) [inline client/components/MovieGrid.js:22]: fetch & set movieData

File: client/index.js [hunk 5]
```diff
@@ -2,14 +2,24 @@ import React from 'react';
...
 
 function App() {
 
   return (
-    <div>HELLO</div>
+    <div>
+      <h2><b>Movie App</b></h2>
+        <div>
+          <TextField id="outlined-basic" label="Search Movies" variant="outlined" />
+          <Button onClick={()=>{console.log("Search movies")}} style={{"background-color":"purple", "margin-left": "10px", "height":"55px"}}> <ArrowForwardIosIcon /> </Button>
```
Comment (fkotsian) [inline client/index.js:17]: 👍 semantic-ui Icons as well: https://semantic-ui.com/elements/icon.html

File: client/components/MovieGrid.js (hunk 3 above)
Comment (fkotsian) [reply client/components/MovieGrid.js:30]: passes down fetched data to components

**Review by fkotsian — APPROVED** [review]
Looks good to me dude! 👍 

The only thing is I forgot to mention that we use Semantic-UI on our legacy projects internally - if you wanted to move this over now I thought it might help get familiar with it, if not I don't think it affects the exercise much. My bad on that 😁

File: client/components/Grid.js (hunk 1 above)
Comment (reshav-abraham) [reply client/components/Grid.js:4]: Will checkout semantic ui. Personally not such a huge fan of ag-grid lol
